src/Bindings/Plugin.h

```diff
@@ -76,8 +76,8 @@ class cPlugin
 	virtual bool OnPlayerBrokenBlock        (cPlayer & a_Player, int a_BlockX, int a_BlockY, int a_BlockZ, char a_BlockFace, BLOCKTYPE a_BlockType, NIBBLETYPE a_BlockMeta) = 0;
 	virtual bool OnPlayerDestroyed          (cPlayer & a_Player) = 0;
 	virtual bool OnPlayerEating             (cPlayer & a_Player) = 0;
-	virtual bool OnPlayerFished             (cPlayer & a_Player, const cItems & a_Reward) = 0;
-	virtual bool OnPlayerFishing            (cPlayer & a_Player, cItems & a_Reward) = 0;
+	virtual bool OnPlayerFished             (cPlayer & a_Player, const cItems & a_Reward, const int ExperienceAmount) = 0;
+	virtual bool OnPlayerFishing            (cPlayer & a_Player, cItems & a_Reward, int ExperienceAmount) = 0;
```

Comment:
This int actually needs to be an int reference, as the point of this hook is to allow the plugin to change the ExperienceAmount (so it shouldn't be a const(ant) parameter). It needs to be a ref (&) so that changes made inside the hook apply to the parameter of the call, rather than the internal copy.

Reply:
Ah I see that makes sense. But const can still be used for the OnPlayerFished, right?

Reply:
Yep, that's the ideal type to use there. you can't change the experience given after it's already happened! A regular const int (not a reference) like you put is right.